### PR TITLE
FIX: pluck device ids before performing the db update.

### PR DESCRIPTION
**Pull request by vinothkannans**
After updating the records `pluck` method won't return the device id of the records which are affected by the `update_all` method.

### CODE REVIEW

File: spec/kolide_spec.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    user.save_custom_fields
+
+    stub_api(:devices)
+    stub_api(:checks)
+    stub_api(:issues)
+  end
+
+  describe "sync!" do
+    it "updates alert post after issues resolved" do
+      issue
```
Comment (janzenisaac) [inline spec/kolide_spec.rb:31]: Do you need to call the fabricator here?

Comment (vinothkannans) [reply spec/kolide_spec.rb:31]: Yes. Else the `issue` won't be created until the next call.

File: spec/kolide_spec.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  before do
+    user.custom_fields["kolide_person_id"] = "23456"
+    user.save_custom_fields
+
+    stub_api(:devices)
+    stub_api(:checks)
+    stub_api(:issues)
+  end
+
+  describe "sync!" do
```
Comment (tgxworld) [inline spec/kolide_spec.rb:29]: Are we supposed to be testing for `.sync_open!` instead since that was where the changes have been made.

Comment (tgxworld) [reply spec/kolide_spec.rb:29]: Ahh my bad this is `Kolid.sync!` and not `Issue.sync!`

**Review by tgxworld — APPROVED** [review]
Changes look good to me.

**Pull request merged**